.github/workflows/benchmark.yaml

```diff
@@ -1,94 +1,103 @@
-    - uses: actions/checkout@v4
-    - name: Download benchmark results
-      uses: actions/download-artifact@v4
+    - id: 'step-0'
+      uses: 'actions/checkout@v4'
+    - id: 'step-1'
+      name: 'Download benchmark results'
+      uses: 'actions/download-artifact@v4'
       with:
-        pattern: bench-results-*
-        path: ${{ env.RESULTS_DIR }}
-        merge-multiple: true
-    - name: Prepare and join benchmark reports
-      id: prep
-      run: |
-        for report in $(find ./${{ env.RESULTS_DIR }} -type f -name "*.json")
-        do
-          file_name=$(basename "$report")
-          platform="${file_name%.*}"
-          # Trim 'it.krzeminski.snakeyaml.engine.kmp.benchmark.' to make benchmark name more readable
-          jq "[ .[] | .benchmark |= \"${platform}.\" + ltrimstr(\"it.krzeminski.snakeyaml.engine.kmp.benchmark.\") | .params |= map_values(. |= match(\"data.+\"; \"g\").string) ]" $report > ${{ env.RESULTS_DIR }}/$platform.json
-        done
-        AGGREGATED_REPORT=aggregated.json
-        # Joined reports looks like this: [[{},{}], [{},{}]]
-        # We need to transform them into this: [{},{}]
-        ls ${{ env.RESULTS_DIR }}/*.json
-        jq -s '[ .[] | .[] ]' ${{ env.RESULTS_DIR }}/*.json > $AGGREGATED_REPORT
-        echo "report=$AGGREGATED_REPORT" >> $GITHUB_OUTPUT
-    - name: Store benchmark result
-      uses: benchmark-action/github-action-benchmark@v1
+        path: 'bench-results'
+        pattern: 'bench-results-*'
+        merge-multiple: 'true'
+    - id: 'step-2'
+      name: 'Prepare and join benchmark reports'
+      env:
+        GHWKT_GITHUB_CONTEXT_JSON: '${{ toJSON(github) }}'
+      run: 'GHWKT_RUN_STEP=''collect-benchmarks-results:step-2'' ''.github/workflows/benchmark.main.kts'''
+    - id: 'step-3'
+      name: 'Store benchmark result'
+      uses: 'benchmark-action/github-action-benchmark@v1'
       with:
-        name: SnakeKMP benchmarks
+        name: 'SnakeKMP benchmarks'
         tool: 'jmh'
-        output-file-path: ${{ steps.prep.outputs.report }}
-        comment-on-alert: true
-        summary-always: true
+        output-file-path: 'aggregated.json'
+        gh-repository: 'github.com/krzema12/snakeyaml-engine-kmp-benchmarks'
+        summary-always: 'true'
+        comment-on-alert: 'true'
         alert-threshold: '150%'
         fail-threshold: '200%'
-        gh-repository: github.com/krzema12/snakeyaml-engine-kmp-benchmarks
-        github-token: ${{ secrets.PUBLISH_BENCHMARK_RESULTS }}
-        # Push and deploy GitHub pages branch automatically only if run in main repo and not in PR
-        auto-push: ${{ github.repository == 'krzema12/snakeyaml-engine-kmp' && github.event_name != 'pull_request' }}
+        github-token: '${{ secrets.PUBLISH_BENCHMARK_RESULTS }}'
+        auto-push: '${{ github.repository == ''krzema12/snakeyaml-engine-kmp'' && github.event_name != ''pull_request'' }}'
```

Comment:
question: Is it okay that the line has two `'` quotes around string values?

Reply:
Yes, it's a way to escape single quotes in a string if we use single quotes for the sting in YAML. It's handled entirely by the library, and "just works" 😊